src/bootstrap/src/lib.rs

```diff
@@ -672,6 +671,7 @@ impl Build {
                 let _sanity_check_span =
                     span!(tracing::Level::DEBUG, "(1) executing dry-run sanity-check").entered();
                 self.config.dry_run = DryRun::SelfCheck;
```

Comment:
Ideally, `dry_run` should be private, and Config should have a method for modifying the dry run value, which would also set the corresponding `exec_ctx` attribute.